src/lookup.py

```diff
@@ -687,162 +700,362 @@ def matches(self, entitytype=None, variant=None, subtype=None):
 
             return True
 
-    def __init__(self, version, subVer, parent):
+    class GroupConfig:
+        def __init__(self, node=None, name=None, label=None, isTab=False):
+            self.iconSize = None
+            self.hasEntity = False
+            if node is not None:
+                self.label = node.get("Label")
+                self.name = node.get("Name", self.label)
+                self.isTab = node.get("IsTab") == "1"
+
+                iconSize = node.get("IconSize")
+                if iconSize:
+                    parts = list(map(lambda x: x.strip(), iconSize.split(",")))
+                    if len(parts) == 2 and checkNum(parts[0]) and checkNum(parts[1]):
+                        self.iconSize = (int(parts[0]), int(parts[1]))
+                    elif len(parts) == 1 and checkNum(parts[0]):
+                        self.iconSize = (int(parts[0]), int(parts[0]))
+                    else:
+                        printf(
+                            f"Group {node.attrib} has invalid IconSize, must be 1 or 2 integer values"
+                        )
+            else:
+                self.name = name
+                self.label = label
+                self.isTab = isTab
+
+            self.entries = []
+            self.groupentries = []
+
+        def addEntry(self, entry):
+            self.entries.append(entry)
+            if isinstance(entry, EntityLookup.GroupConfig):
+                self.groupentries.append(entry)
+            elif isinstance(entry, EntityLookup.EntityConfig):
+                self.hasEntity = True
+
+        def getAllEntries(self):
+            entries = []
+            for entry in self.entries:
+                entries.append(entry)
+                if isinstance(entry, EntityLookup.GroupConfig):
+                    entries.extend(entry.getAllEntries())
+
+            return entries
+
+        def containsEntity(self):
+            if self.hasEntity:
```

Comment:
should probably make it clearer in the names what the intended use case for each boolean is. Or just make hasEntity private